File: back/src/main/java/com/openclassrooms/mddapi/dto/ErrorResponse.java
package com.openclassrooms.mddapi.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.Instant;
import java.util.Map;

@Data
@AllArgsConstructor
public class ErrorResponse {
    private Instant timestamp;
    private int status;
    private String error;
    private String message;
    private Map<String, String> fieldErrors;
}

File: front/src/app/core/http/error-response.ts
/** Mirror of the backend `ErrorResponse` DTO. */
export interface ErrorResponse {
  /** ISO-8601 UTC instant (suffix `Z`). */
  timestamp: string;
  status: number;
  error: string;
  message: string;
  /** Present on validation errors only; `null` otherwise. */
  fieldErrors: Record<string, string> | null;
}
